### PR TITLE
update controller-gen

### PR DESCRIPTION
**Pull request by kim-tsao**
Signed-off-by: Kim Tsao <ktsao@redhat.com>

**Please specify the area for this PR**

**What does does this PR do / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #?
https://github.com/devfile/api/issues/1013

In addition to the changes mentioned in this [comment](https://github.com/devfile/api/issues/1013#issuecomment-1416085552), I had to update the controller-manager and kube-rbac-proxy sccs in order to fix the warning on openshift:

`Warning: would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (container "kube-rbac-proxy" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (containers "kube-rbac-proxy", "manager" must set securityContext.capabilities.drop=["ALL"]), seccompProfile (pod or containers "kube-rbac-proxy", "manager" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")`

**PR acceptance criteria**:

- [ ] Test Coverage 
    - Are your changes sufficiently tested, and are any applicable test cases added or updated to cover your changes?
- [ ] Gosec scans 
  - Fix all MEDIUM and higher findings and/or annotate a finding per gosec instructions: https://github.com/securego/gosec#annotating-code to address why a finding is not a security issue

Documentation
- [ ] Does the [registry operator documentation](https://github.com/devfile/devfile-web/tree/main/libs/docs/src/docs/no-version) need to updated with your changes?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Comment by michael-valdron** [comment]
`go mod tidy: go.mod file indicates go 1.18, but maximum supported version is 1.17` our integration tests need updating to support golang 1.18, we could cap it at 1.17 until this support is added.

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM golang:1.17 as builder
+FROM golang:1.18 as builder
```
Comment (michael-valdron) [inline Dockerfile:2]:
```suggestion
FROM golang:1.17 as builder
```
> `go mod tidy: go.mod file indicates go 1.18, but maximum supported version is 1.17` our integration tests need updating to support golang 1.18, we could cap it at 1.17 until this support is added.

Keep it at 1.17 for now.

File: go.mod [hunk 2]
```diff
@@ -1,21 +1,104 @@
 module github.com/devfile/registry-operator
 
-go 1.13
+go 1.18
```
Comment (michael-valdron) [inline go.mod:3]:
```suggestion
go 1.17
```
> `go mod tidy: go.mod file indicates go 1.18, but maximum supported version is 1.17` our integration tests need updating to support golang 1.18, we could cap it at 1.17 until this support is added.

Update it to golang 1.17 instead for now.

**Review by michael-valdron — CHANGES_REQUESTED** [review]
Aside from my suggestions there might be other changes needed to stay at golang 1.17.

**Review by michael-valdron — APPROVED** [review]
> `go mod tidy: go.mod file indicates go 1.18, but maximum supported version is 1.17` our integration tests need updating to support golang 1.18, we could cap it at 1.17 until this support is added.

/lgtm

I see your changes for testing support https://github.com/openshift/release/pull/36276 you can disregard my review suggestions.

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Comment by kim-tsao** [comment]
Also updated a couple of dependencies related to upgrading the operator-sdk: https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.25.0/

**Pull request merged**